pyscf/grad/test/test_tduks_grad.py

```diff
@@ -49,18 +50,18 @@ def tearDownModule():
 
 class KnownValues(unittest.TestCase):
     def test_tda_lda(self):
-        td = tdscf.TDA(mf_lda).run(nstates=3)
+        td = tdscf.TDA(mf_lda).run(nstates=nstates)
         tdg = td.nuc_grad_method()
         g1 = tdg.kernel(td.xy[2])
-        self.assertAlmostEqual(g1[0,2], -0.40279473514282405, 6)
+        self.assertAlmostEqual(g1[0,2], -0.7944872119457362, 6)
```

Comment:
@sunqm requesting more states (5 rather than 3) seems to be the cause. For example, even using the master branch which has `nstates=3` gives the following excitation energies
```
[0.12166646 0.12166646 0.80854428]
```
while changing it to `nstates=5` gives
```
[0.12166646 0.12166646 0.75668038 0.80854428 0.80854428]
```
You can see that the energy of the third state changes. In general, I found it numerically more stable to request a few more states. So I changed it to `nstates=5` in almost all tdscf-related tests. Not sure if this is a general solution or there's anything we can do on the davidson solver side.